src/analyzer.ts

```diff
@@ -52,6 +55,14 @@ function isDependencyExternal(url: string) {
   return parseUrl(url).protocol !== null || url.startsWith('//');
 }
 
+/**
+ * Get a longer, single-line error message for logging and exeption-handling
+ * analysis Warning objects.
```

Comment:
Hmm, creating a new stream that uses our logger just for analysis seems a little round-a-bout, passing a function would be an easier interface to use as a consumer.

Will create a seperate issue to discuss.